gateway-service/src/main/java/org/zowe/apiml/gateway/conformance/ConformanceProblemsContainer.java

```diff
@@ -0,0 +1,148 @@
+    }
+
+    @Override
+    public int size() {
+        int result = 0;
+        for (ArrayList<String> value : this.values()) {
+            if (value == null) {
+                continue;
+            }
+            result += value.size();
+        }
+        return result;
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder result = new StringBuilder();
+
+        result.append("{");
+
+        boolean firstLoop = true;
+
+        ArrayList<String> sortedKeySet = new ArrayList<>(this.keySet());
+        sortedKeySet.remove(null);  // since it used be a set this removes all nulls
+        sortedKeySet.sort(null);
+
+        for (String key : sortedKeySet) {
+
+            if (this.get(key) == null || this.get(key).size() == 0) {
+                continue;
+            }
+
+
+            if (!firstLoop) {
+                result.append(",");
+            }
+
+
+            result.append("\"").append(key).append("\"");
+            result.append(":[");
+
+            boolean firstInnerLoop = true;
+            for (String i : get(key)) {
+                if (!firstInnerLoop) {
+                    result.append(",");
+                }
+                try {
+                    result.append(new ObjectMapper().writeValueAsString(i));
+                } catch (JsonProcessingException e) {
+                    throw new RuntimeException(e);
+                }
+                firstInnerLoop = false;
+            }
+            result.append("]");
+
+            firstLoop = false;
+        }
+
+        result.append("}");
+        return result.toString();
+    }
+
+
+    public String createBadRequestAPIResponseBody(String key, ApiMessage correspondingAPIMessage) {
+        String result;
+
+        String template = "{\n" +
+            "    \"messageAction\": \"replaceWithMessageAction\",\n" +
+            "    \"messageContent\": {\n" +
+            "        \"The service " + serviceId + " is not conformant\": \n" +
+            "            replaceWithMessageContent\n" +
+            "    },\n" +
+            "    \"messageKey\": \"replaceWithMessageKey\",\n" +
+            "    \"messageNumber\": \"replaceWithMessageNumber\",\n" +
+            "    \"messageReason\": \"replaceWithMessageReason\",\n" +
+            "    \"messageType\": \"replaceWithMessageType\"\n" +
+            "}";
+
+        result = template.replace("replaceWithMessageKey", key);
+        result = result.replace("replaceWithMessageContent", this.toString());
+
+        result = result.replace("replaceWithMessageReason", correspondingAPIMessage.getMessageReason());
+        result = result.replace("replaceWithMessageNumber", correspondingAPIMessage.getMessageNumber());
+        result = result.replace("replaceWithMessageType", correspondingAPIMessage.getMessageType().toString());
+        result = result.replace("replaceWithMessageAction", correspondingAPIMessage.getMessageAction());
+
```

Comment:
check `String.format()` or better `java.text.MessageFormat.format` for named variables